executor/include/execution/trait/is_executor.hpp

```diff
@@ -0,0 +1,27 @@
+//
+// Created by Shrijit Singh on 14/06/20.
+//
+
+#pragma once
+
+#include <experimental/type_traits>
+#include <type_traits>
+
+namespace execution {
+
+template <class Executor, typename = void>
+struct is_executor : std::false_type {};
+
+constexpr const auto noop = [] {};
+
+template <class Executor>
+struct is_executor<
+    Executor,
+    std::enable_if_t<
+        std::is_same_v<void, decltype(std::declval<Executor>().execute(noop))>,
```

Comment:
With this you're saying that: any class which defines an executor method that accepts an invocable argument and returns void is an executor. I suspect you will be returning arbitrary types from your execute method.

Aren't you only interested in:
  1. checking is there's an execute method which accepts an invocable argument? (ignoring completely the return type)
  2. checking that the Executor is derived from the base executor class

Reply:
This is the latest definition of an executor according to `p1660r0`
```
an Executor should be a
CopyConstructible and EqualityComparable type that provides a function named execute that eagerly submits work on a single execution agent created for it by the executor
```

I will be switching to this definition in a separate PR.

```cpp
// An Executor is something with an .execute()
// member function that takes a nullary callable.
template<class Ex, class C = void(*)()>
concept Executor =
Invocable<C> &&
CopyConstructible<remove_cvref_t<Ex>> &&
EqualityComparable<remove_cvref_t<Ex>> &&
requires (Ex&& ex, C&& c) {
((Ex&&) ex).execute((C&&) c);
};
```